standards/application/dag-ack.md

```diff
@@ -0,0 +1,155 @@
+- If the `d1` did not exist in store node, charlie broadcasts a `request` message in the group.
+- Dave gets the `request` message and check if `d1` exists in local or not. 
+- Dave has `d1` and broadcast it to the network.
+
+
+## Payload Format
+
+### Tip Message
+
+The references for a message and checkpoint messages are generally encrypted in `WakuMessage.payload`. The first reference is always the previous message from the sender. The following references are the latest messages from other group members. The payload for `Tip Message` is defined as following,
+
+```python
+class Payload:
+    def __init__(self, references):
+        # omit other attributes
+        self.references = references
+```
+
+For test purpose, it can also be encoded in `WakuMessage.meta` for public analysis in following manner,
+
+```python
+concat(message_hash_1, message_hash_2, message_hash_3)
+```
+
+### Checkpoint Message
+
+The payload for `Checkpoint Message` is similar to Tip Message, but it only contains references, no other attributes.
+
+### Request Message
+
+The payload for `Request Message` is defiled as following,
+
+```python
+class Payload:
+    def __init__(self, request_hash):
+        self.request_hash = request_hash
+```
+
+## Principle
+
+### Choose Reference
+
+The algorithm to choose references is basically randomly but give active users higher probability. Online users are generally seen as active. If app doesn't have online/offline capability, it's acceptable to say the user who has sent out any messages in the last `n` (for example, 7) days is treated as active. Below is a recommend implementation in Python, 
+
+```python
+import random
+
+def weighted_choose(group1, group2, prob_group1, k):
+    combined_group = group1 + group2
+    weights = [prob_group1 / len(group1)] * len(group1) + [(1 - prob_group1) / len(group2)] * len(group2)
+
+    # Ensure k does not exceed the number of unique elements
+    k = min(k, len(set(combined_group)))
+
+    # Generate a weighted choice list ensuring no duplicates
+    selected = []
+    while len(selected) < k:
+        choice = random.choices(combined_group, weights, k=1)[0]
+        if choice not in selected:
+            selected.append(choice)
+    
+    return selected
+
+# Example usage
+active_group = ['Alice', 'Bob', 'Charlie']
+inactive_group = ['Dave', 'Eve', 'Frank']
+prob_active = 0.7  # 70% chance of picking a member from active group
+k = 3  # Number of unique members to pick
+
+picked_members = weighted_choose(active_group, inactive_group, prob_active, k)
+
+print("Picked members:", picked_members)
+```
+
+**Note**: It's recommended to always include a reference to your previous message, and two or three references to other group members. If the group is small (less than 5 members), one reference to other members is acceptable. If the group is large and the length of message payload is not a limitation, following curve can be considered,
+
+```
+f = a * log(x)
```

Comment:
Could you explain a bit more how this curve would work for very large groups? For Community protocol our eventual aim is to support Communities of up to 10K members. Presumably this would create a significant bandwidth and processing overhead?